src/Microsoft.OData.Core/UriParser/Visitors/IsCollectionTranslator.cs

```diff
@@ -157,5 +193,17 @@ public override bool Translate(MetadataSegment segment)
             ExceptionUtils.CheckArgumentNotNull(segment, "segment");
             return false;
         }
+
+        /// <summary>
+        /// Translate a PathTemplateSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>UserDefinedValue</returns>
+        /// <exception cref="System.ArgumentNullException">Throws if the input segment is null.</exception>
+        public override bool Translate(PathTemplateSegment segment)
+        {
+            ExceptionUtils.CheckArgumentNotNull(segment, "segment");
+            return false;

```

Comment:
>return false [](start = 12, length = 12)

maybe we should return segment.SingleResult (in case we someday figure out how to determine whether the path template is single- or multi-valued.)

Reply:
Changed to `return !segment.SingleResult;`